lib/DxilValidation/DxilValidation.cpp

```diff
@@ -2756,6 +2756,44 @@ static void ValidateFunctionBody(Function *F, ValidationContext &ValCtx) {
             }
             dispatchMesh = CI;
           }
+
+          if (dxilOpcode == DXIL::OpCode::AllocateRayQuery) {
+            // validate flags are immediate and compatible
+            llvm::Value *constRayFlag = CI->getOperand(1);
+            if (!llvm::isa<llvm::Constant>(constRayFlag)) {
+              ValCtx.EmitInstrError(
+                  &I, ValidationRule::
+                          DeclNonConstFlagsUnsupportedForAllocateRayQuery);
+            }
+          }
+          if (dxilOpcode == DXIL::OpCode::AllocateRayQuery2) {
```

Comment:
>if (dxilOp [](http://example.com/codeflow?start=10&length=10)

nit: formatting looks like there should be a blank line before every if

Reply:
It does, but looks like our code formatting bot doesn't like it for some reason, reverting.